src/Stryker.CLI/Stryker.CLI/ExitCodes.cs

```diff
@@ -0,0 +1,18 @@
+namespace Stryker.CLI
+{
+    /// <summary>
+    /// Known Stryker exit codes.
+    /// </summary>
+    public static class ExitCodes
+    {
+        /// <summary>
+        /// Stryker is executed against .NET Framework but no solution file is given.
+        /// </summary>
+        public static int NoSolutionFileSpecified = 1;
```

Comment:
Is this actually what we use exit code 1 for? I would think exit code 1 is used for anything that throws an exception?

Reply:
Well not sure actually, empirically that's the only case where I've seen 1 (besides violating threshold break which is set explicitly). For example, [here](https://github.com/stryker-mutator/stryker-net/issues/1659) it's `-532462766` (I plan to add it when addressing that issue).

Reply:
Afaik we do not set any specific exit code for no solution specified, we simply throw a stryker input exception: https://github.com/stryker-mutator/stryker-net/blob/master/src/Stryker.Core/Stryker.Core/Initialisation/InitialBuildProcess.cs#L36 which we eventually catch here: https://github.com/stryker-mutator/stryker-net/blob/master/src/Stryker.CLI/Stryker.CLI/Program.cs#L22 where we return exit code 1.

Reply:
Oh right thaaaat's why it's 1 there. Makes sense. 

I reshuffled this a bit then. Maybe let it be `ErrorCodes` not `ExitCodes` for now as 0 is also technically an exit code but codifying that would require changing a lot things in the codebase and I didn't want to mix refactoring with logical changes in the PR.

Reply:
I would love to know what changes would be needed to support ExitCode 0, because the only place I know about is right here: https://github.com/stryker-mutator/stryker-net/blob/master/src/Stryker.CLI/Stryker.CLI/StrykerCLI.cs#L27

If it required a lot of effort to incorporate exit 0 then I agree that ErrorCodes is better for now.

Reply:
Well technically I would imagine this to be replaced everywhere we test it, so 6 more places as per this:
![image](https://user-images.githubusercontent.com/5451366/130623097-0955d7a1-b693-4517-86e4-171fd94d743e.png)

So yeah depends on how clean you want to have your git history. I would be happy to make this as a follow up, but I can do it in this PR as well.

Reply:
Ah right, didn't think of those. Yea if you want make this change go ahead I don't mind!

Reply:
Alright, hang on then :)

Reply:
Alright, so I think I've updated all the places in the source. I moved ExitCodes from the CLI to Core to make this possible, hope that's okay.

Reply:
The move feels like we're combining two concepts into one here, because one is the stryker exit codes and the other is exit codes of other apps. If we don't combine these concepts then the stryker exit codes could remain with stryker.cli, since stryker.core does not really 'exit' that makes sense to me. However it is also nice to give a name to the magic numbers used in stryker.core. I think it's fine now, there's little benefit other than clear split concerns by splitting them again. Looks good to me 👍

Reply:
Yeah we can later reshuffle things around again. Thanks!